build-system/pr-check/ci-job.js

```diff
@@ -57,6 +58,22 @@ async function runCiJob(jobName, pushBuildWorkflow, prBuildWorkflow) {
   }
 }
 
+/**
+ * Returns list of test files that CircleCI should execute in a parallelized job shard.
+ *
+ * @param {!Array<string>} globs array of glob strings for finding test file paths.
+ * @return {string} a comma separated list of test file paths.
+ */
+function getCircleCiShardTestFiles(globs) {
```

Comment:
This is clever, thanks for figuring it out. I have two high-level comments about this function:
1. The intent behind `ci-job.js` was to be a minimal skeleton that's common across all CI jobs. Since this is a helper that's used by some jobs but not others, a better place for it would be a helper file like `build-system/pr-check/utils.js`.
2. As discussed yesterday, `--files` is mainly used for local development, and its implementation differs from what we use during CI (logging, error handling, choice of reporters, etc.). Also, it has the downside of [very long logging messages](https://user-images.githubusercontent.com/26553114/126356886-d2984bf1-085a-487b-972e-e171eff96b85.png) that make the logs hard to read. I suggest switching this function to write the list of test suites to disk, and adding a new flag called `--filelist` that reads the file passed in. (This can be done by adding a new function `getFilesFromFilelist()` to `build-system/common/utils.js`, similar to `getFilesFromArgv()` that reads the list).

WDYT?

Reply:
Added commit 20de51b to address (1) and 4c1b552 to address (2), PTAL

Reply:
Also wdyt about `--from_file` for the flag name? (reaching  because this is expected to only be used in CI

Reply:
Also wdyt about --from_file for the flag name? (reaching 🚲🏚 moment because this is expected to only be used in CI)

Reply:
Thanks for making this change. I think the flag ought to be `--filelist` (not `--fileslist`) because it is a well-known standard concept / name. (E.g. Google search [redirects](https://www.google.com/search?q=FilesList) `FilesList` to `FileList` 😃 )

Reply:
Done.